### PR TITLE
Implement BufferAPI: BufferConfig

### PR DESCRIPTION
**Pull request by 123mpozzi**
## Description

Implement BufferConfig to control player buffer depth

## Changes

- Add `BufferMediaTypeConfig`, `BufferConfig`
- Extend `PlayerConfig` interface to include a bufferConfig property
- Add jsdoc links to mentioned objects inside `PlayerConfig`

## Checklist
- [x] 🗒 `CHANGELOG` entry

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 - `PictureInPictureConfig` to `PlayerView` to allow configuring the Picture in Picture behavior
 - `PictureInPictureConfig.shouldEnterOnBackground` to start PiP automatically when the app transitions to background
 - `onPictureInPictureAvailabilityChanged` event is now emitted on iOS and tvOS in addition to Android
+- `MediaType`, `BufferType`, `BufferLevel`, `BufferMediaTypeConfig`, `BufferConfig` to have types and interfaces necessary for incoming `BufferAPI`
```
Comment (rolandkakonyi) [inline CHANGELOG.md:11]:
```suggestion
- `BufferConfig` to control player buffer depth
```

File: src/player.ts [hunk 2]
```diff
@@ -66,6 +67,10 @@ export interface PlayerConfig extends NativeInstanceConfig {
    * Configures remote playback functionality.
    */
   remoteControlConfig?: RemoteControlConfig;
+  /**
+   * Configures buffer settings. A default {@link BufferConfig} is set initially
```
Comment (rolandkakonyi) [inline src/player.ts:71]:
```suggestion
   * Configures buffer settings. A default {@link BufferConfig} is set initially.
```

**Review by rolandkakonyi — COMMENTED** [review]
Changes look good!

I will manually test it next.

**Review by rolandkakonyi — APPROVED** [review]
Manual testing looks good too!

**Review by matamegger — APPROVED** [review]
👍

**Pull request merged**